private/buf/cmd/buf/command/registry/sdk/version/version.go

```diff
@@ -0,0 +1,225 @@
+	flags := newFlags()
+	return &appcmd.Command{
+		Use:   name + " --module=<buf.build/owner/repository[:ref]> --plugin=<buf.build/owner/plugin[:version]>",
+		Short: "Resolve module and plugin reference to a specific Generated SDK version",
+		Long: `This command returns the version of the Generated SDK for the given module and plugin.
+Examples:
+
+Get the version of the eliza module and the go plugin for use with the Go module proxy.
+    $ buf registry sdk version --module=buf.build/connectrpc/eliza --plugin=buf.build/protocolbuffers/go
+    v1.33.0-20230913231627-233fca715f49.1
+
+Use a specific module version and plugin version.
+    $ buf registry sdk version --module=buf.build/connectrpc/eliza:233fca715f49425581ec0a1b660be886 --plugin=buf.build/protocolbuffers/go:v1.32.0
+    v1.32.0-20230913231627-233fca715f49.1`,
+		Args: appcmd.NoArgs,
+		Run: builder.NewRunFunc(
+			func(ctx context.Context, container appext.Container) error {
+				return run(ctx, container, flags)
+			},
+		),
+		BindFlags: flags.Bind,
+	}
+}
+
+// TODO FUTURE: Add a --format flag, supports text (current behavior) and json, json will output information
+// such as resolved module commit, plugin version and revision, package-ecosystem, and full version.
+
+type flags struct {
+	Plugin string
+	Module string
+}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {
+	flagSet.StringVar(&f.Module, moduleFlagName, "", "The module reference to resolve")
+	flagSet.StringVar(&f.Plugin, pluginFlagName, "", "The plugin reference to resolve")
+	_ = appcmd.MarkFlagRequired(flagSet, moduleFlagName)
+	_ = appcmd.MarkFlagRequired(flagSet, pluginFlagName)
+}
+
+func run(
+	ctx context.Context,
+	container appext.Container,
+	flags *flags,
+) error {
+	moduleRef, err := bufmodule.ParseModuleRef(flags.Module)
+	if err != nil {
+		return appcmd.NewInvalidArgumentErrorf(err.Error())
+	}
+	pluginIdentity, pluginVersion, err := bufpluginref.ParsePluginIdentityOptionalVersion(flags.Plugin)
+	if err != nil {
+		return appcmd.NewInvalidArgumentErrorf(err.Error())
+	}
+	if pluginIdentity.Remote() != moduleRef.ModuleFullName().Registry() {
+		return appcmd.NewInvalidArgumentError("module and plugin must be from the same BSR instance")
+	}
+
+	clientConfig, err := bufcli.NewConnectClientConfig(container)
+	if err != nil {
+		return err
+	}
+	pluginCurationServiceClient := connectclient.Make(
+		clientConfig,
+		moduleRef.ModuleFullName().Registry(),
+		registryv1alpha1connect.NewPluginCurationServiceClient,
+	)
+	resolveServiceClient := connectclient.Make(
+		clientConfig,
+		moduleRef.ModuleFullName().Registry(),
+		registryv1alpha1connect.NewResolveServiceClient,
+	)
+	getLatestCuratedPluginResponse, err := pluginCurationServiceClient.GetLatestCuratedPlugin(
+		ctx,
+		connect.NewRequest(
+			&registryv1alpha1.GetLatestCuratedPluginRequest{
+				Owner:   pluginIdentity.Owner(),
+				Name:    pluginIdentity.Plugin(),
+				Version: pluginVersion,
```

Comment:
@pkwarren on second thought, I don't think we should support the revision here. Given an explicit version we should always return the latest revision no matter what. WDYT?

Reply:
In the API or in this command? I looked some more and getting the plugin revision would be more complex if we wanted to support it in this command. We explicitly broke it out into a separate yaml field so that users would get the latest revision.

I don't mind it in the API - the UI has a drop down for revision and may need to display different data based on the revision.

Reply:
Sorry, I meant in this command. I thought you mentioned wanting to support revision, and my only comment was to advise against it specifically in this command.

Reply:
Agree - I don't think we should support it here.